File: frontend/src/types/exhibitionReadiness.ts
// ExhibitionReadinessReason 展览公开展出阻断原因。
export interface ExhibitionReadinessReason {
  code: string;
  message: string;
  count: number;
  artworkIds: string[];
}

// ExhibitionReadiness GET /api/v1/exhibitions/:id/readiness 响应。
// 按关联作品实时统计，任何异常都作为阻断原因。
export interface ExhibitionReadiness {
  exhibitionId: string;
  exhibition?: {
    id: string;
    title: string;
    status: string;
    reviewStatus: string;
  };
  conclusion: string;
  ready: boolean;
  isEmpty: boolean;
  total: number;
  readyCount: number;
  pendingCount: number;
  soldCount: number;
  archivedCount: number;
  noImageCount: number;
  readyRatio: number;
  reasons: ExhibitionReadinessReason[];
}

File: backend/internal/model/exhibition_readiness.go
package model

// ReadinessReasonCode 展览就绪阻断原因码。
type ReadinessReasonCode string

// ExhibitionReadiness 展览公开展出就绪统计（按关联作品实时计算）。
type ExhibitionReadiness struct {
	ExhibitionID  string
	Exhibition    *Exhibition
	Total         int
	ReadyCount    int
	PendingCount  int
	SoldCount     int
	ArchivedCount int
	NoImageCount  int
	ReadyRatio    float64
	Ready         bool
	IsEmpty       bool
	Reasons       []ReadinessBlockReason
}

// ReadinessBlockReason 单个阻断原因及命中作品。
type ReadinessBlockReason struct {
	Code       ReadinessReasonCode
	Message    string
	Count      int
	ArtworkIDs []string
}
